src/Stratis.Bitcoin/Configuration/TextFileConfiguration.cs

```diff
@@ -122,12 +123,14 @@ public void MergeInto(TextFileConfiguration destination)
         /// </summary>
         /// <param name="key">Name of the argument (excluding the dash prefix).</param>
         /// <returns>Values for the specified argument.</returns>
-        public string[] GetAll(string key)
+        public string[] GetAll(string key, ILogger logger = null)
```

Comment:
Is it not simpler to create the `TextFileConfiguration` with the logger inside it (if available)?

Reply:
That's the point. We wan't to have logs for the namespace of the caller

Reply:
Ah you mean each settings ns ok fair enough

Reply:
It does look cleaner indeed.